include/range/v3/utility/iterator_concepts.hpp

```diff
@@ -497,55 +497,56 @@ namespace ranges
                 meta::compose<
                     meta::uncurry<meta::on<ReduceFn, meta::uncurry<MapFn>>>,
                     meta::quote<iter_args_lists_>>;
+
+            template<template <typename...> class InvocableConcept, typename C, typename ...Is>
+            using indirect_invocable_ = meta::and_<
+                meta::strict_and<Readable<Is>...>,
+                // C must satisfy the InvocableConcept with the values and references read from the Is.
+                meta::lazy::invoke<
+                    iter_map_reduce_fn_<
+                        meta::bind_front<meta::quote<InvocableConcept>, C&>,
+                        meta::quote<meta::strict_and>>,
+                    Is...>>;
+            
+            template<typename C, typename ...Is>
+            using common_result_indirect_invocable_ = meta::and_<
+                indirect_invocable_<Invocable, C, Is...>,
+                // In addition to C being invocable, the return types of the C invocations must all
+                // share a common reference type. (The lazy::invoke is so that this doesn't get
+                // evaluated unless C is truly callable as determined above.)
+                meta::lazy::invoke<
+                    iter_map_reduce_fn_<
+                        meta::bind_front<meta::quote<concepts::Invocable::result_t>, C&>,
+                        meta::quote<CommonReference>>,
+                    Is...>>;
         }
 
         template<typename C, typename ...Is>
         using IndirectInvocable = meta::and_<
+            detail::common_result_indirect_invocable_<C, Is...>,
             CopyConstructible<uncvref_t<C>>,
```

Comment:
Yes, I looked at this from this perspective, but thought maybe I'm missing something - so I left it as it was. I'll fix it.